LS_Miroshnichenko/main.swift

```diff
@@ -8,5 +8,35 @@
 
 import Foundation
 
-print("Hello, World!")
+let clHelper = CommandLineIOHelper()
+var userArguments = clHelper.input()
 
+if userArguments.count < 2 {
+    clHelper.output(outputText: "parameter is not defined, open help with parameter \"-h\"", outputType: .error)
+    exit(0)
+}
+
+if userArguments.contains("-h") {
+    clHelper.help()
+    exit(0)
+}
+
+let fileManager = FileManager.default
+let currenDirectoryPath = fileManager.currentDirectoryPath
+let shouldSkipHiddenFiles = !userArguments.contains("-a")
+
+guard let urls = fileManager.urls(for: currenDirectoryPath, skipsHiddenFiles: shouldSkipHiddenFiles) else {
+    exit(0)
+}
+
+let shouldPrintAttributes = userArguments.contains("-l")
+
+if shouldPrintAttributes {
+    for url in urls {
+        clHelper.output(outputText: "\(AttributesDescription(attributes: url.attributes))   \(url.relativePath)", outputType: .message)
+    }
+} else {
+    for url in urls {
+        clHelper.output(outputText: "\(url.relativePath)", outputType: .message)
+    }
+}
```

Comment:
minor stylistic comment:
it'd be neater to check `shouldPrintAttributes` inside a loop to avoid copypasted loop in the code. like this:
```suggestion

for url in urls {
    let text = shouldPrintAttributes ? "\(AttributesDescription(attributes: url.attributes))   \(url.relativePath)" :
                                       "\(url.relativePath)"
    clHelper.output(outputText: text, outputType: .message)
}

```
Also you can use forEach instead of for loop, but this is up to you

Reply:
Is it Ok to check the `shouldPrintAttributes` at each iteration? 
And if the check would come with a difficult condition computation? Is there an even more elegant way?

Reply:
That's a good point! Here we have a trade-off between readability and performance. Avoiding one spare condition is good, but in this particular case the condition doesn't provide any significant performance penalty. If the condition is not just one bool value, but something that needs to be calculated, then we should do the calculations before the loop and use result bool value inside. If the condition depends on each array element, we're not able to check it outside the loop anyway. 
I don't see any alternatives to having 2 cycles or having condition inside of one (although it doesn't mean there's none :) )